modules/blazingsql/src/blazingcluster.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2021, NVIDIA CORPORATION.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {DataFrame, TypeMap} from '@rapidsai/cudf';
+import {ChildProcess, fork} from 'child_process';
+
+import {UcpContext} from './addon';
+import {BlazingContext} from './blazingcontext';
+
+export const CREATE_BLAZING_CONTEXT = 'createBlazingContext';
+export const CREATE_TABLE           = 'createTable';
+export const RUN_QUERY              = 'runQuery';
+export const QUERY_RAN              = 'ranQuery';
+export const CONFIG_OPTIONS         = {
+  PROTOCOL: 'UCX',
+};
+
+interface BlazingCusterProps {
+  numWorkers: number;
+}
+
+export class BlazingCluster {
+  workers: ChildProcess[];
+  bc: BlazingContext;
+
+  constructor({numWorkers = 1}: BlazingCusterProps) {
+    this.workers = Array(numWorkers).fill(fork(`${__dirname}/worker`, {serialization: 'advanced'}));
```

Comment:
won't this fill all the slots of the `workers` Array with the same ChildProcess instance?